src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -0,0 +1,1112 @@
+                }
+                else
+                {
+                    // #10593: Add More Tests for Serialization Code
+                    Debug.Assert(false);
+                }
+            }
+            else if (o is byte[])
+            {
+                byte[] a = (byte[])o;
+                if (method == "WriteElementStringRaw")
+                {
+                    WriteElementStringRaw(name, ns, FromByteArrayBase64(a));
+                }
+                else if (method == "WriteNullableStringLiteralRaw")
+                {
+                    WriteNullableStringLiteralRaw(name, ns, FromByteArrayBase64(a));
+                }
+                else if (method == "WriteAttribute")
+                {
+                    WriteAttribute(name, ns, a);
+                }
+                else
+                {
+                    // #10593: Add More Tests for Serialization Code
+                    Debug.Assert(false);
+                }
+            }
+            else
+            {
+                // #10593: Add More Tests for Serialization Code
+                Debug.Assert(false);
+            }
+
+        }
+
+        private bool IsDefaultValue(TypeMapping mapping, object o, object value, bool isNullable)
+        {
+            if (value is string && ((string)value).Length == 0)
+            {
+                string str = (string)o;
+                return str == null || str.Length == 0;
+            }
+            else
+            {
+                return value.Equals(o);
+            }
+        }
+
+        private bool WritePrimitiveValue(TypeDesc typeDesc, object o, bool isElement, out string stringValue)
+        {
+            if (typeDesc == StringTypeDesc || typeDesc.FormatterName == "String")
+            {
+                stringValue = (string)o;
+                return true;
+            }
+            else
+            {
+                if (!typeDesc.HasCustomFormatter)
+                {
+                    stringValue = CovertPrimitiveToString(o, typeDesc);
+                    return true;
+                }
+                else if (o is byte[] && typeDesc.FormatterName == "ByteArrayHex")
+                {
+                    stringValue = FromByteArrayHex((byte[])o);
+                    return true;
+                }
+                else if (o is DateTime && typeDesc.FormatterName == "DateTime")
+                {
+                    stringValue = FromDateTime((DateTime)o);
+                    return true;
+                }
+                else if (o is DateTime && typeDesc.FormatterName == "Date")
+                {
+                    stringValue = FromDate((DateTime)o);
+                    return true;
+                }
+                else if (o is DateTime && typeDesc.FormatterName == "Time")
+                {
+                    stringValue = FromTime((DateTime)o);
+                    return true;
+                }
```

Comment:
Check if o is DateTime once, then check the typeDesc.FormatterName in a switch block inside.